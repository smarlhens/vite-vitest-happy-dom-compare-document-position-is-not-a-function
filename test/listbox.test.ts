import type { RenderResult } from '@testing-library/vue';
import {fireEvent, render} from '@testing-library/vue';
import { afterEach, describe, expect, it } from 'vitest';

import Listbox from '../src/components/listbox.vue';

describe('listbox.vue', () => {
  let wrapper: RenderResult;
  const mountOptions = {
    props: {
      modelValue: 20,
    },
  };

  afterEach(() => {
    wrapper.unmount();
  });

  it('should render', async () => {
    wrapper = render(Listbox, mountOptions);

    const { getByText, html } = wrapper;
    const button = getByText('Wade Cooper');
    await fireEvent.click(button);
    expect(html()).toMatchSnapshot();
  });


  it('should trigger an event on size change', async () => {
    wrapper = render(Listbox, mountOptions);

    const { getByText, findByText } = wrapper;
    const button = getByText('Wade Cooper');
    await fireEvent.click(button);

    const option = await findByText('Arlene Mccoy');
    await fireEvent.click(option);
    await findByText('Arlene Mccoy');
  });
});
