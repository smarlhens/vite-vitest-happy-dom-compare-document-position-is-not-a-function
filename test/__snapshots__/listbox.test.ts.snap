// Vitest Snapshot v1

exports[`listbox.vue > should render 1`] = `
"<div class=\\"fixed w-72\\" modelvalue=\\"20\\">
  <div class=\\"relative mt-1\\"><button id=\\"headlessui-listbox-button-1\\" type=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\" class=\\"relative w-full cursor-default rounded-lg bg-white py-2 pl-3 pr-10 text-left shadow-md focus:outline-none focus-visible:border-indigo-500 focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75 focus-visible:ring-offset-2 focus-visible:ring-offset-orange-300 sm:text-sm\\"><span class=\\"block truncate\\">Wade Cooper</span><span class=\\"absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none\\"><svg class=\\"h-5 w-5 text-gray-400\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M10 3a1 1 0 01.707.293l3 3a1 1 0 01-1.414 1.414L10 5.414 7.707 7.707a1 1 0 01-1.414-1.414l3-3A1 1 0 0110 3zm-3.707 9.293a1 1 0 011.414 0L10 14.586l2.293-2.293a1 1 0 011.414 1.414l-3 3a1 1 0 01-1.414 0l-3-3a1 1 0 010-1.414z\\" clip-rule=\\"evenodd\\"></path></svg></span></button>
    <transition-stub>
      <ul aria-labelledby=\\"headlessui-listbox-button-1\\" aria-orientation=\\"vertical\\" id=\\"headlessui-listbox-options-3\\" role=\\"listbox\\" tabindex=\\"0\\" class=\\"absolute mt-1 max-h-60 w-full overflow-auto rounded-md bg-white py-1 text-base shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none sm:text-sm\\">
        <li class=\\"text-gray-900 relative cursor-default select-none py-2 pl-10 pr-4\\" id=\\"headlessui-listbox-option-4\\" role=\\"option\\" aria-selected=\\"true\\"><span class=\\"font-medium block truncate\\">Wade Cooper</span><span class=\\"text-amber-600 absolute inset-y-0 right-0 flex items-center pr-4\\"><svg class=\\"h-5 w-5\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z\\" clip-rule=\\"evenodd\\"></path></svg></span></li>
        <li class=\\"text-gray-900 relative cursor-default select-none py-2 pl-10 pr-4\\" id=\\"headlessui-listbox-option-5\\" role=\\"option\\"><span class=\\"font-normal block truncate\\">Arlene Mccoy</span>
          <!--v-if-->
        </li>
        <li class=\\"text-gray-900 relative cursor-default select-none py-2 pl-10 pr-4\\" id=\\"headlessui-listbox-option-6\\" role=\\"option\\"><span class=\\"font-normal block truncate\\">Devon Webb</span>
          <!--v-if-->
        </li>
        <li class=\\"text-gray-900 relative cursor-default select-none py-2 pl-10 pr-4\\" id=\\"headlessui-listbox-option-7\\" role=\\"option\\"><span class=\\"font-normal block truncate\\">Tom Cook</span>
          <!--v-if-->
        </li>
        <li class=\\"text-gray-900 relative cursor-default select-none py-2 pl-10 pr-4\\" id=\\"headlessui-listbox-option-8\\" role=\\"option\\"><span class=\\"font-normal block truncate\\">Tanya Fox</span>
          <!--v-if-->
        </li>
        <li class=\\"text-gray-900 relative cursor-default select-none py-2 pl-10 pr-4\\" id=\\"headlessui-listbox-option-9\\" role=\\"option\\"><span class=\\"font-normal block truncate\\">Hellen Schmidt</span>
          <!--v-if-->
        </li>
      </ul>
    </transition-stub>
  </div>
</div>"
`;
